deployment/README.md

```diff
@@ -0,0 +1,92 @@
+# GCP Project Setup (WIP)
+
+Ensure the `GOOGLE_CLOUD_PROJECT` environment variable is exported for the project:
+```bash
+export GOOGLE_CLOUD_PROJECT=<PROJECT_ID>
+```
+The official project id of OSV is `oss-vdb`.
+
+## Terraform
+
+In this directory (`/deployment`):
+
+Initialise terraform:
+```bash
+terraform init
+```
+
+Plan and apply required project infastructure:
+```bash
+terraform plan
+```
+```bash
+terraform apply
+```
+Use these commands to deploy any configuration changes.
+Inspect the proposed changes when running the command to see what resources will
+be added, modified and destroyed.
+
+## Setting up auto-scaler
+
+There doesn't seem to be a good way to set this up within terraform.
+
+The following instructions are from [here](https://cloud.google.com/kubernetes-engine/docs/tutorials/external-metrics-autoscaling#step1).
+
+```bash
+kubectl create clusterrolebinding cluster-admin-binding \
+    --clusterrole cluster-admin --user "$(gcloud config get-value account)"
+```
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/GoogleCloudPlatform/k8s-stackdriver/master/custom-metrics-stackdriver-adapter/deploy/production/adapter_new_resource_model.yaml
```

Comment:
This is different from the one we actually use in our production project:

```
https://raw.githubusercontent.com/GoogleCloudPlatform/k8s-stackdriver/master/custom-metrics-stackdriver-adapter/deploy/production/adapter.yaml
```

Is there a meaningful difference between the two?

Reply:
I didn't notice that. The [linked steps](https://cloud.google.com/kubernetes-engine/docs/tutorials/autoscaling-metrics#pubsub_1) point to using the new resource model for pubsub, rather than that legacy one.

From the [GitHub page](https://github.com/GoogleCloudPlatform/k8s-stackdriver/tree/master/custom-metrics-stackdriver-adapter#configure-cluster):
> Stackdriver supports two models of Kubernetes resources: **the legacy model**
using monitored resource `gke_container` and **the new model** using different
Kubernetes monitored resources, including for example `k8s_pod`, `k8s_node`. See
[monitored resources documentation](https://cloud.google.com/monitoring/api/resources) for more details.

I don't know what actual difference this makes, but I can change this to the legacy one if that would be better.

Reply:
Did you verify if the autoscaling works? I think @another-rex tried the non-legacy one a while back on the production project and it didn't work. 

Fine to keep this as is for now, just flagging that this may need a bit more investigation.

Reply:
Noted.
It did scale down from 4 nodes to 1 when I activated it, and I don't think it's spat out any errors. I've not tried publishing anything to see if it scales up, though.